### PR TITLE
allow user defined setting for retain-story-order button in ControlPanel

### LINKED ISSUE #6848: [Proposal] decouple "retain story order" button from translatable texts with a new config tiddler
## The Problem

While working on a new plugin to create an "Empty Story River" backdrop view, that allows users to click the "Home Button" I found the following problem. 

To reproduce.

- open https://tiddlywiki.com
- Open the https://tiddlywiki.com/#%24%3A%2FControlPanel
- Click the "retain story ordering" button in the DefaultTiddlers section

So your defaultTiddlers should look like this: `[list[$:/StoryList]]` --> Everything is OK. 

- Now Close all tiddlers --> Story list is empty. 
- Click -- Right Sidebar : Tools : Home Button ----> nothing happens ---> go on reading

The function is OK, but changing the behaviour of the "retain story ordering" button is hard for plugin authors because

[$:/language/ControlPanel/Basics/DefaultTiddlers/BottomHint](https://github.com/Jermolene/TiddlyWiki5/blob/e849aaab4db951e62089ac57484b1592ea89802a/core/language/en-GB/ControlPanel.multids#L10) ... Is hardcoded and translatable. So If I would like to change the behaviour of the button, I need to overwrite this core tiddler, and I loose the language switching possibility for that button.

The existing code is: 

```
Use &#91;&#91;double square brackets&#93;&#93; for titles with spaces. Or you can choose to <$button set="$:/DefaultTiddlers" setTo="[list[$:/StoryList]]">retain story ordering</$button>
```

## The proposed code is:

```
Use &#91;&#91;double square brackets&#93;&#93; for titles with spaces. Or you can choose to <$button set="$:/DefaultTiddlers" setTo={{$:/config/ControlPanel/Basics/DefaultTiddlers/setTo}}>retain story ordering</$button>
```

Advantages

 - Everyone can change the behaviour of the button, without loosing all translations
 - Plugin authors can change the button to set defaultTiddlers to eg: `[list[$:/StoryList]] ~GettingStarted` 
   - The setting above will open the GettingStarted tiddler if the StoryList is empty
 - Since some pages may have a different tiddler that may be shown eg: tiddlywiki.com ... It could be set to `[list[$:/StoryList]] ~HelloThere` 

The only thing that's needed is a new config tiddler, which may be part of the HiddenSettings

@Jermolene 
A PR that adjusts all ControlPanel.multids files for all languages can follow if you are willing to merge it.

### PR DESCRIPTION
**Pull request by pmario**
This PR fixes #6848 and allows users to set the default configuration for the "retain story order" button in the ControlPanel : Info : Basics area.

- The PR contains a new doc tiddler for the hidden setting, 
- the changes to ControlPanel.multids
- the button-snippet tiddler 
- and the config tiddler. 

~~This is a draft, since changes or all existing languages are missing. I'll extend the missing changes when you are OK with the changes~~

### CODE REVIEW

File: editions/tw5.com/tiddlers/hiddensettings/Hidden Setting Default Story Ordering.tid [hunk 1]
```diff
@@ -0,0 +1,4 @@
+title: Hidden Setting: Retain Story ordering
+tags: [[Hidden Settings]]
+
+<<.from-version "5.2.23">> The tiddler $:/config/ControlPanel/Basics/DefaultTiddlers/RetainStory defaults to `[list[$:/StoryList]]` and enables users and plugin authors to change the default value, that the "retain story ordering" button sets to the $:/DefaultTiddlers in the $:/ControlPanel : Basics : Info tab.
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/hiddensettings/Hidden Setting Default Story Ordering.tid:4]: Could be better phrased as:

```
<<.from-version "5.2.23">> The tiddler $:/config/ControlPanel/Basics/DefaultTiddlers/RetainStory contains the value that is assigned to $:/DefaultTiddlers when clicking the "retain story ordering" button in $:/ControlPanel under the ''Info'' -> ''Basics'' tab.
```

**Review by Jermolene — COMMENTED** [review]
Thanks @pmario

A minor comment – we should really try to be consistent with how we refer to control panel tabs; the convention here of using colons to separate the tab names doesn't appear to be used elsewhere in the docs. Really important to be consistent

**Comment by pmario** [comment]
@Jermolene ... I did add the other languages. I did a visual check but an other pair of eyes would be good :)

@kookma .. Can you please check the Persian language. 
@BramChen .. Please check the Chinese languages. 
@xcazin ... Please check French
@twMat ... Can you have a look too?
@TiddlyTweeter ... Pls have a look ... Italian?

and all other translators, who are watching ... please check your language

**Comment by xcazin** [comment]
> @xcazin ... Please check French 

@pmario I confirm that both English and French versions work perfectly, with the very same look and functionality than in the master branch.

**Comment by BramChen** [comment]
> @BramChen .. Please check the Chinese languages.

Thanks @pmario, I do check chinese editions, they look work fine.

**Comment by twMat** [comment]
> @twMat ... Can you have a look too?

I'd be happy to, but I need to ask a general question (sorry for the OT): Is the request that I read the code under the "Files changed" here in the PR? Or is there some way to see/try the rendered result - and if there is, how do I actually get PR code into a TW? I don't see a gh feature to download PRs.

**Comment by pmario** [comment]
>I'd be happy to, but I need to ask a general question (sorry for the OT): Is the request that I read the code under the "Files changed" here in the PR?

It's not OT. ... I was generally asking to review the code here at GH, to have more eyes on the code, since I did change every language file that exists. ... I replaced `<$button set="$:/DefaultTiddlers" setTo="[list[$:/StoryList]]">retain story ordering</$button>` with `{{retain story ordering||$:/snippets/retain-story-ordering-button}}` 

Since I can only read German and English, It is always good to have some reviewers, that especially check the languages they know. .. And have a fast look on the other languages as well, if there looks something strange. 

>Or is there some way to see/try the rendered result - and if there is, how do I actually get PR code into a TW? I don't see a gh feature to download PRs.

You would need to know git to download this branch of the code and build a language version. ... If that's not your thing. A review here at GH will be fine. 

I'm pretty sure, I got it right. But -- The more eyes look at it the better.

**Comment by twMat** [comment]
@pmario - OK then, it looks good. 

BUT, unrelated to your changes, I did find a the [Swedish translation](https://github.com/Jermolene/TiddlyWiki5/pull/6863/files#diff-082349721abddd085491fcdcea570d8bdd477f92d1ce8e2a4c15ee9db18cd71e) to be poorly worded so I'm thinking I should do a general review of the whole Swedish translation. I posted #6876 for further discussion.

**Comment by pmario** [comment]
bump

File: core/ui/ControlPanel/BasicsRetainStoryOrderingSnippet.tid [hunk 2]
```diff
@@ -0,0 +1,3 @@
+title: $:/snippets/retain-story-ordering-button
```
Comment (Jermolene) [inline core/ui/ControlPanel/BasicsRetainStoryOrderingSnippet.tid:1]: The title of this tiddler does not match the filename

File: editions/tw5.com/tiddlers/hiddensettings/Hidden Setting Default Story Ordering.tid [hunk 3]
```diff
@@ -0,0 +1,4 @@
+title: Hidden Setting: Retain Story ordering
+tags: [[Hidden Settings]]
+
+<<.from-version "5.2.23">> The tiddler $:/config/ControlPanel/Basics/DefaultTiddlers/RetainStory contains the value that is assigned to $:/DefaultTiddlers when clicking the "retain story ordering" button in $:/ControlPanel under the ''Info'' -> ''Basics'' tab.
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/hiddensettings/Hidden Setting Default Story Ordering.tid:4]: I think this needs to be updated to 5.2.24

**Review by Jermolene — COMMENTED** [review]
Thanks @pmario a couple of points

**Comment by pmario** [comment]
@Jermolene .. did fix the typos

**Comment by pmario** [comment]
@Jermolene pls merge

**Comment by Jermolene** [comment]
Thanks @pmario

**Pull request merged**